cache/refs.go

```diff
@@ -631,109 +628,116 @@ func (sr *immutableRef) ociDesc(ctx context.Context, dhs DescHandlers) (ocispecs
 }
 
 const (
-	compressionVariantDigestLabelPrefix      = "buildkit.io/compression/digest."
-	compressionVariantAnnotationsLabelPrefix = "buildkit.io/compression/annotation."
-	compressionVariantMediaTypeLabel         = "buildkit.io/compression/mediatype"
+	blobVariantGCLabel         = "containerd.io/gc.ref.content.blob-"
+	blobAnnotationsLabelPrefix = "buildkit.io/blob/annotation."
+	blobMediaTypeLabel         = "buildkit.io/blob/mediatype"
 )
 
-func compressionVariantDigestLabel(compressionType compression.Type) string {
-	return compressionVariantDigestLabelPrefix + compressionType.String()
-}
-
-func getCompressionVariants(ctx context.Context, cs content.Store, dgst digest.Digest) (res []compression.Type, _ error) {
-	info, err := cs.Info(ctx, dgst)
-	if errors.Is(err, errdefs.ErrNotFound) {
-		return nil, nil
-	} else if err != nil {
-		return nil, err
-	}
-	for k := range info.Labels {
-		if strings.HasPrefix(k, compressionVariantDigestLabelPrefix) {
-			if t := compression.Parse(strings.TrimPrefix(k, compressionVariantDigestLabelPrefix)); t != compression.UnknownCompression {
-				res = append(res, t)
-			}
-		}
-	}
-	return
-}
-
-func (sr *immutableRef) getCompressionBlob(ctx context.Context, compressionType compression.Type) (ocispecs.Descriptor, error) {
-	return getCompressionVariantBlob(ctx, sr.cm.ContentStore, sr.getBlob(), compressionType)
-}
-
-func getCompressionVariantBlob(ctx context.Context, cs content.Store, dgst digest.Digest, compressionType compression.Type) (ocispecs.Descriptor, error) {
-	info, err := cs.Info(ctx, dgst)
-	if err != nil {
-		return ocispecs.Descriptor{}, err
-	}
-	dgstS, ok := info.Labels[compressionVariantDigestLabel(compressionType)]
-	if ok {
-		dgst, err := digest.Parse(dgstS)
-		if err != nil {
-			return ocispecs.Descriptor{}, err
-		}
-		return getBlobDesc(ctx, cs, dgst)
-	}
-	return ocispecs.Descriptor{}, errdefs.ErrNotFound
-}
-
-func (sr *immutableRef) addCompressionBlob(ctx context.Context, desc ocispecs.Descriptor, compressionType compression.Type) error {
+func (sr *immutableRef) addBlob(ctx context.Context, desc ocispecs.Descriptor) error {
```

Comment:
Should this method also set the ref's `size` metadata  to `sizeUnknown` so that future calls to `sr.size()` recalculate the new size from the addition of a new blob?

Reply:
Thank you for the suggestion. Fixed to set `size` to `sizeUnknown`.